src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt

```diff
@@ -15,17 +19,102 @@ class Personfakta(private val datagrunnlag: Datagrunnlag) {
 
     fun personensDokumenterIJoark(): List<Journalpost> = datagrunnlag.dokument
 
-    fun personensSisteStatsborgerskap(): String = datagrunnlag.personhistorikk.statsborgerskap[0].landkode
+    fun personensSisteStatsborgerskap(): String {
+        val periodeDatagrunnlag = Interval.of(datagrunnlag.periode.fom.atStartOfDay(ZoneId.systemDefault()).toInstant(), datagrunnlag.periode.tom.atStartOfDay(ZoneId.systemDefault()).toInstant())
+        val statsborgerskapIPeriode = datagrunnlag.personhistorikk.statsborgerskap.filter {
+            periodeDatagrunnlag.overlaps(lagInterval(Periode(it.fom, it.tom))) ||
+            periodeDatagrunnlag.encloses(lagInterval(Periode(it.fom, it.tom)))
+        }
+
+        statsborgerskapIPeriode.forEach{
```

Comment:
return statsborgerskapIPeriode.find { !eøsLand.containsKey(it.landkode) } ?: statsborgerskapIPeriode.last().landkode